pkg/cloudprovider/aws/instance.go

```diff
@@ -172,7 +172,12 @@ func (p *InstanceProvider) getLaunchTemplateConfigs(ctx context.Context, constra
 func (p *InstanceProvider) getOverrides(instanceTypeOptions []cloudprovider.InstanceType, subnets []*ec2.Subnet, capacityType string) []*ec2.FleetLaunchTemplateOverridesRequest {
 	var overrides []*ec2.FleetLaunchTemplateOverridesRequest
 	for i, instanceType := range instanceTypeOptions {
-		for zone := range instanceType.Zones() {
+		for _, offering := range instanceType.Offerings() {
+			zone := offering.Zone
```

Comment:
optional: I might just inline this where its used. I don't think there's much of a perf angle, and it adds an extra concept to think about as you read.

Reply:
Yeah, this was an artifact of the uniqueness check